plugins/lookup/bitwarden_secrets_manager.py

```diff
@@ -84,11 +85,28 @@ class BitwardenSecretsManagerException(AnsibleLookupError):
 class BitwardenSecretsManager(object):
     def __init__(self, path='bws'):
         self._cli_path = path
+        self._max_retries = 3
+        self._retry_delay = 1
 
     @property
     def cli_path(self):
         return self._cli_path
 
+    def _run_with_retry(self, args, stdin=None, retries=0):
+        if retries > self._max_retries:
+            raise BitwardenSecretsManagerException("Max retries exceeded. Unable to retrieve secret.")
+
+        out, err, rc = self._run(args, stdin)
+
+        if "Too many requests" in err:
```

Comment:
Shouldn't it also check `rc` here? Or does the CLI tool return with `rc == 0` in case of rate limit failures?

Reply:
[14c6074](https://github.com/ansible-collections/community.general/pull/8238/commits/14c607414979a34e55510a365040d769990fa4ee) Should help in making it more clear that it's a non-zero exit condition